esmvalcore/preprocessor/_regrid.py

```diff
@@ -1423,7 +1419,11 @@ def get_reference_levels(dataset):
 
 
 @preserve_float_dtype
-def extract_coordinate_points(cube, definition, scheme):
+def extract_coordinate_points(
+    cube: Cube,
+    definition: dict[str, ArrayLike],
+    scheme: str,
```

Comment:
Use a `Literal` to limit the accepted values to valid ones, e.g. `scheme: Literal["linear", "nearest"]`

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2772/commits/d7c73aa172b3a91d172080d60f55bb5467263531